src/dmtcp_restart.cpp

```diff
@@ -471,10 +471,27 @@ runMtcpRestart(int is32bitElf, int fd, ProcessInfo *pInfo)
     }
   }
 
+  /* If DMTCP_RESTART_PAUSE==1, mtcp_restart will loop until gdb attach.*/
+  bool mtcp_restart_pause = false;
+  char * pause_param = getenv("DMTCP_RESTART_PAUSE");
+  if (pause_param == NULL) {
+    pause_param = getenv("MTCP_RESTART_PAUSE");
+  }
+  if (pause_param != NULL && pause_param[0] == '1' && pause_param[1] == '\0') {
```

Comment:
I suppose we could write `&& strncmp(pause_param, "1", 2)` instead of `&& pause_param[0] == '1' && pause_param[1] == '\0'`.  But I find the use of `strncmp()` less readable.  It seems to say to compare the first two characters of a string of length one: "1".  It works because of a trick based on knowing that a string of length one has an internal representation of length two: `{'1', '\0'}`.
Normally, people use `strncmp` when they want to use `strcmp` and are worried about running off the end of a large buffer.  We could avoid relying on knowledge of the internal string representation by writing `strcmp(pause_param, "1")` since we're not worried about the end of some buffer.  But I find the C primitives more intuitive for a string of length one.